storage/src/mmr/journaled.rs

```diff
@@ -307,6 +312,7 @@ impl<E: RStorage + Clock + Metrics, H: CHasher> Mmr<E, H> {
             for pos in Proof::<H::Digest>::nodes_to_pin(cfg.lower_bound) {
                 let digest =
                     Mmr::<E, H>::get_from_metadata_or_journal(&metadata, &journal, pos).await?;
+                metadata.put(U64::new(NODE_PREFIX, pos), digest.to_vec());
```

Comment:
This seems like the important line.  Are you sure the other two lines above adding nodes to metadata are needed?  (although you would have to make sure metadata gets the pinned nodes if cfg.lower_bound == journal_size.)

Usually what you want to do is compute the pinned nodes required by the journaled MMR, and make sure those are in metadata.  The mem-mmr's pinned nodes don't need to be in the metadata since they are either already in the metadata, or part of journal's regular node set.

Reply:
(I guess I don't fully understand the case where pinned nodes are provided though.)